### PR TITLE
Bug 1788253: fix deployment config flake by using informer

### PR DESCRIPTION
**Pull request by mfojtik**
Attempt to address https://bugzilla.redhat.com/show_bug.cgi?id=1788253

### CODE REVIEW

File: test/extended/deployments/deployments.go [hunk 1]
```diff
@@ -1651,11 +1658,35 @@ var _ = g.Describe("[Feature:DeploymentConfig] deploymentconfigs", func() {
...
+						Group:    "apps.openshift.io/v1",
+						Resource: "DeploymentConfig",
+					}, dc.Name)
+				}
+
+				return false, nil
+			}
+			lw := &cache.ListWatch{
+				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+					return oc.AppsClient().AppsV1().DeploymentConfigs(oc.Namespace()).List(metav1.ListOptions{})
```
Comment (tnozicka) [inline test/extended/deployments/deployments.go]: why are you dropping options?

File: test/extended/deployments/deployments.go [hunk 2]
```diff
@@ -1550,8 +1554,11 @@ var _ = g.Describe("[Feature:DeploymentConfig] deploymentconfigs", func() {
 			rcs, err := oc.KubeClient().CoreV1().ReplicationControllers(namespace).List(metav1.ListOptions{
 				LabelSelector: appsutil.ConfigSelector(dc.Name).String(),
 			})
-			o.Expect(rcs.Items).To(o.HaveLen(1))
-			o.Expect(strings.TrimSpace(rcs.Items[0].Spec.Template.Spec.Containers[0].Image)).NotTo(o.BeEmpty())
+			o.Expect(rcs).NotTo(o.BeNil())
```
[the remark this answers is not in the document]
Comment (mfojtik) [reply test/extended/deployments/deployments.go]: just missing and goland complained, unrelated to anything

File: test/extended/deployments/deployments.go (hunk 1 above)
Comment (mfojtik) [reply test/extended/deployments/deployments.go]: do we need options here? does it make a difference?

File: test/extended/deployments/deployments.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tnozicka) [reply test/extended/deployments/deployments.go]: I mean this is missing the check for err, would you mind adding it? Goland will also stop complaing if you add `o.Expect(err).NotTo(o.HaveOccurred())` and you don't need the nil check

File: test/extended/deployments/deployments.go [hunk 3]
```diff
@@ -1624,12 +1631,34 @@ var _ = g.Describe("[Feature:DeploymentConfig] deploymentconfigs", func() {
 			// When a DC is recreated it has LatestVersion 0, it will get updated after adopting the Rcs
 			o.Expect(dc.Status.LatestVersion).To(o.BeEquivalentTo(0))
 
+			dcListWatch := &cache.ListWatch{
```
Comment (tnozicka) [inline test/extended/deployments/deployments.go]: fieldSelector := fields.OneTermEqualSelector("metadata.name", dc.Name).String()

File: test/extended/deployments/deployments.go [hunk 4]
```diff
@@ -1624,12 +1631,34 @@ var _ = g.Describe("[Feature:DeploymentConfig] deploymentconfigs", func() {
...
+				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+					return oc.AppsClient().AppsV1().DeploymentConfigs(oc.Namespace()).List(options)
```
Comment (tnozicka) [inline test/extended/deployments/deployments.go]: `options.FieldSelector = fieldSelector`

Comment (tnozicka) [reply test/extended/deployments/deployments.go]: nit: s/oc.Namespace()/namespace/

Comment (mfojtik) [reply test/extended/deployments/deployments.go]: fixed.

File: test/extended/deployments/deployments.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mfojtik) [reply test/extended/deployments/deployments.go]: gotcha, fixed.

**Pull request merged**